### PR TITLE
Implement remote contract events support

### LINKED ISSUE #1344: handle events in remote contracts
as we recently noticed there is currently no way for the SDK to handle events of remote contracts.

thanks to @thepiwo we now at least ignore unknown events after this PR:
- https://github.com/aeternity/aepp-sdk-js/pull/1337/files

we should investigate this and find a way how to handle that situation. for the dex implementation we want to be able to write tests that depend on events emitted from a remote contract.

IMO we should at least be able to handle this if we know the ACI of all contracts involved in the interaction.

### PR DESCRIPTION
**Pull request by davidyuk**
depends on https://github.com/aeternity/aepp-calldata-js/pull/112
depends on #1395 
closes #1344 except for events with the same name in different contracts

### CODE REVIEW

**Comment by marc0olo** [comment]
that's awesome! =)

do we have any idea to handle same names?

I didn't check in detail yet how it works with this PR. do we rely on any additional http request or is it fully covered with calldata-lib in the sdk?

**Comment by davidyuk** [comment]
> do we have any idea to handle same names?

nope

> do we rely on any additional http request or is it fully covered with calldata-lib in the sdk?

the events before were decoded by sdk, without requests to http compiler or so, in this PR I've removed our poor implementation and switched to calldata

File: test/integration/contract-aci.js [hunk 1]
```diff
@@ -520,7 +543,7 @@ describe('Contract instance', function () {
 
       it('Invalid arguments length', async () => {
         await expect(testContract.methods.datTypeGFn())
-          .to.be.rejectedWith('Non matching number of arguments. Got 0 but expected from 1 to 1')
+          .to.be.rejectedWith('Non matching number of arguments. datTypeGFn expects between 1 and 1 number of arguments but got 0')
```
Comment (thepiwo) [inline test/integration/contract-aci.js:546]: weird error message

**Review by thepiwo — APPROVED** [review]
pretty good if I understand correctly

**Comment by davidyuk** [comment]
> do we have any idea to handle same names?

I've added `contractAddressToName` option to resolve this

**Review by marc0olo — APPROVED** [review]
lgtm

File: test/integration/contract-aci.js (hunk 1 above)
Comment (marc0olo) [reply test/integration/contract-aci.js:628]: @davidyuk does this message come from calldata-lib ? just wondering

**Review by thepiwo — APPROVED** [review]

**Pull request merged**